flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/BaseReconciler.java

```diff
@@ -75,29 +83,44 @@ public boolean removeDeployment(FlinkDeployment flinkApp) {
                                 flinkApp.getMetadata().getName(),
                                 flinkApp.getMetadata().getNamespace());
                         jobManagerDeployments.add(flinkApp.getMetadata().getUid());
-                        if (flinkApp.getStatus().getJobStatus() != null) {
-                            // pre-existing deployments on operator restart - proceed with
-                            // reconciliation
-                            return null;
-                        }
+                        return JobDeploymentStatus.READY;
```

Comment:
The above check was required to delay job check to not run into timeout while skipping this for pre-existing deployment. How is that covered now?

Reply:
JobDeploymentStatus.READY means that the deployment and ports are ready. In that case there should not be any timeout. Did I overlook something?

Reply:
There is a period of typically a few seconds between port ready and rest server ready and if you hit that then the list jobs will run into a timeout. And hence it is also necessary to make a distinction for existing deployments and operator restart, where that isn't relevant. I think I will start covering these scenarios with test cases since we cannot afford to continue moving things around w/o regression checks.

Reply:
I will try to make sure this case is accounted for with a test case.